packages/pwa-kit-extension-sdk/src/shared/utils/extensibility.ts

```
@@ -92,19 +92,27 @@ export const findFileWithExtension = (basePath: string, extensions: string[] = [
 }
 
 /**
- * This regular expression is used to identify and validate application extensions.
- * The current formate for a valid extension is an optional package namespace followed by
- * `extension-` string litteral, then the extension name. (E.g. @salesforce/extension-store-finder).
- * This regex has grouping that allows you to pull the namespace and application extension
- * name from the string being matched.
+ * Extensions are identified by the presence of an `extension-meta.json` file in the package directory.
+ * This method allows us to find extensions without relying on any naming convention.
  *
  * @example
  *
  * const installedExtensions = Object.keys(pkg.devDependencies)
- *      .map((packageName) => packageName.match(nameRegex))
- *      .filter(Boolean)
+ *      .filter((packageName) => isExtensionPackage(path.join(projectDir, 'node_modules', packageName)))
  */
-export const nameRegex = /^(?:@([^/]+)\/)?extension-(.+)$/
+export const isExtensionPackage = (packagePath: string): boolean => {
+    try {
+        if (!packagePath) {
+            return false
+        }
```

What are your thoughts on throwing instead of returning false?

---

I think for the `isExtensionPackage` scenario gracefully handling the invalid input with false, i.e. no packagePath means "not an extension", is better than crashing the application

---

Got it.. the way I was looking at it, is that the path is a required argument, so if one isn't passed it, it represents a dev error and should tho. But there are 2 sides to the coin, see my conversation with ai [here](https://chatgpt.com/share/6818eaa1-7904-800d-b08b-ada343a9f79a). 

I'm ok with either way.